polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -84,14 +88,56 @@ bool isCandidate(LoopLikeOpInterface loop) {
   return true;
 }
 
+/// Get the size of unused shared local memory arena in bytes.
+unsigned getLocalMemRemain(Operation &module) {
```

Comment:
This check is not sufficient to get the remaining local memory of a kernel. 

SYCL has `local_accessor`s for dynamically sized local memory (in that case, the kernel entry point has an address-space 3 pointer), which need to be taken into account when calculating the remaining local memory size. As the size of dynamically sized local memory is only known at runtime (at least without analyzing the host), I think we would need to check the kernel for any dynamically sized local memory and not perform the transformation, if any `local_accessor` is present. 

For more information, see [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.local) and [here](https://link.springer.com/chapter/10.1007/978-1-4842-5574-2_9).

Reply:
done.

Reply:
Created https://github.com/intel/llvm/pull/9713 to not perform the transformation when dynamic sized local accessor is used.